lib/resty/timerng/thread/loop.lua

```diff
@@ -166,6 +169,22 @@ local function loop_wrapper(premature, self)
 
     if not do_phase_handler(self, "init") then
         while not ngx_worker_exiting() and not self._kill do
+            local global_level = _G.log_level
+
+            if global_level then
+                local sys_filter_level = get_sys_filter_level()
```

Comment:
we can probably just store the current log level in `self` and compare with `global level` ? but on the other hand i'm okay to use `get_sys_filter_level`
because it's also in our PDK, if it got changed, lots of tests will make sure
we are aware of it.

Reply:
that's true, we can also probably just store the current log level in the global `_G` table as well 🤔 but yeah I'm also fine with `get_sys_filter_level` ...